---
title: "Compare the DEG and DET lists"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(magrittr)
library(dplyr)
library(data.table)
library(readr)
library(ggVennDiagram)
library(biomaRt)
library(UpSetR)
library(ThreeDRNAseq)
```

# expressed genes

```{r}
RfeatureCounts_gene_results_afterfilter <- readRDS("RfeatureCounts_gene_results_afterfilter.rds")
associated_gene_counts_with_novel_afterfilter <- readRDS("associated_gene_counts_with_novel_afterfilter.rds")
venn_list <- list(isoseq_expressed_gene = rownames(associated_gene_counts_with_novel_afterfilter),
     RNAseq_expressed_gene = rownames(RfeatureCounts_gene_results_afterfilter$counts))

ggVennDiagram(venn_list, label = "count")
```

```{r}
kallisto_transcript_TPM <- readRDS("kallisto_transcript_TPM.rds") %>% as.data.frame()
# filter the TPM
kallisto_transcript_TPM <- kallisto_transcript_TPM[TPM.filter(kallisto_transcript_TPM,sample.n = 3, tpm.cut = 1),]

venn_list$kallisto_expressed_gene  <- rownames(kallisto_transcript_TPM) %>% gsub("\\.[^\\.]+$","",.) %>% unique()

ggVennDiagram(venn_list[c(1,3)], label = "count")
```

# compare between the results of DEG for Iso_seq and RNA_seq
```{r}
setwd("~/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Lloyd_RNAseq_cattle_R")

LIST_DEG_results <- list()

LIST_DEG_results[[1]]  <- read_tsv("IsoSeq_significant_with_novel_DEG_0.05.tsv") %>% subset(regulation %in% "down")
LIST_DEG_results[[2]]  <- read_tsv("IsoSeq_significant_without_novel_DEG_0.05.tsv") %>% subset(regulation %in% "down")

LIST_DEG_results[[3]]  <- read_tsv("RfeatureCounts_significant_with_novel_DEG_0.05.tsv") %>% subset(regulation %in% "down")
LIST_DEG_results[[4]]  <- read_tsv("RfeatureCounts_significant_without_novel_DEG_0.05.tsv") %>% subset(regulation %in% "down")

LIST_DEG_results[[5]] <- read_tsv("sleuth_significant_DEG_0.05.tsv") %>% subset(regulation %in% "FALSE")
```


```{r}
ggVenn_list <- lapply(c(1,2,3,4), function(x){LIST_DEG_results[[x]]%>% extract2("gene_id")})
ggVenn_list[[5]] <- LIST_DEG_results[[5]] %>% extract2("target_id")

names(ggVenn_list) <- c("Isoseq_DEG_Ensembl&Pacbio", "Isoseq_DEG_Ensembl", "RNAseq_DEG_Ensembl&Pacbio", "RNAseq_DEG_Ensembl", "RNAseq_DEG_Ensembl&Pacbio_Kallistio")

ggVennDiagram(ggVenn_list[1:2], label = "count")
ggVennDiagram(ggVenn_list[3:4], label = "count")
```

```{r}
ggVennDiagram(ggVenn_list[c(1,3)], label = "count")
```
# upset plots
```{r}
Upset_DEG <- c(ggVenn_list$`Isoseq_DEG_Ensembl&Pacbio`, ggVenn_list$`RNAseq_DEG_Ensembl&Pacbio`, ggVenn_list$`RNAseq_DEG_Ensembl&Pacbio_Kallistio`) %>% 
  unique() %>% 
  as.data.frame() %>%
  set_colnames("ID")

Upset_DEG$`Isoseq_DEG`[Upset_DEG$ID %in% c(ggVenn_list$`Isoseq_DEG_Ensembl&Pacbio`)] <- 1
Upset_DEG$`Isoseq_DEG`[is.na(Upset_DEG$`Isoseq_DEG`)] <- 0

Upset_DEG$`RNAseq_DEG`[Upset_DEG$ID %in% c(ggVenn_list$`RNAseq_DEG_Ensembl&Pacbio`)] <- 1
Upset_DEG$`RNAseq_DEG`[is.na(Upset_DEG$`RNAseq_DEG`)] <- 0

Upset_DEG$`RNAseq_DEG_Kallistio`[Upset_DEG$ID %in% c(ggVenn_list$`RNAseq_DEG_Ensembl&Pacbio_Kallistio`)] <- 1
Upset_DEG$`RNAseq_DEG_Kallistio`[is.na(Upset_DEG$`RNAseq_DEG_Kallistio`)] <- 0

upset(Upset_DEG, nintersects = 30, mb.ratio = c(0.5, 0.5),main.bar.color = "blue", keep.order = T)

```

# expressed transcript

```{r}
RfeatureCounts_transcript_results_afterfilter <- readRDS("RfeatureCounts_transcript_results_afterfilter.rds")
associated_transcript_counts_with_novel_afterfilter <- readRDS("associated_isoform_counts_with_novel_afterfilter.rds")

venn_list <- list(isoseq_expressed_transcript = rownames(associated_transcript_counts_with_novel_afterfilter),
     RNAseq_expressed_transcript = rownames(RfeatureCounts_transcript_results_afterfilter))

ggVennDiagram(venn_list, label = "count")
```

```{r}
kallisto_transcript_TPM <- readRDS("kallisto_transcript_TPM.rds") %>% as.data.frame()
# filter the TPM
kallisto_transcript_TPM <- kallisto_transcript_TPM[TPM.filter(kallisto_transcript_TPM,sample.n = 3, tpm.cut = 1),]

venn_list$kallisto_expressed_transcript <- rownames(kallisto_transcript_TPM)

ggVennDiagram(venn_list[c(1,3)], label = "count")
```

# compare between the results of DET for Iso_seq and RNA_seq
```{r}
LIST_DET_results <- list()

LIST_DET_results[[1]]  <- read_tsv("IsoSeq_significant_with_novel_DET_0.05.tsv") %>% subset(regulation %in% "up")
LIST_DET_results[[2]]  <- read_tsv("IsoSeq_significant_without_novel_DET_0.05.tsv") %>% subset(regulation %in% "up")

LIST_DET_results[[3]]  <- read_tsv("RfeatureCounts_significant_with_novel_DET_0.05.tsv") %>% subset(regulation %in% "up")
LIST_DET_results[[4]]  <- read_tsv("RfeatureCounts_significant_without_novel_DET_0.05.tsv") %>% subset(regulation %in% "up")

LIST_DET_results[[5]] <- read_tsv("sleuth_significant_DET_0.05.tsv") %>% 
  subset(b > 0)

```

```{r}
ggVenn_list <- lapply(c(1,2,3,4), function(x){LIST_DET_results[[x]]%>% extract2("transcript_id")})
ggVenn_list[[5]] <- LIST_DET_results[[5]] %>% extract2("target_id")

names(ggVenn_list) <- c("Isoseq_DET_Ensembl&Pacbio", "Isoseq_DET_Ensembl", "RNAseq_DET_Ensembl&Pacbio", "RNAseq_DET_Ensembl","RNAseq_DET_Ensembl&Pacbio_Kallistio")
ggVennDiagram(ggVenn_list[1:2], label = "count")
ggVennDiagram(ggVenn_list[3:4], label = "count")
```

```{r}
ggVennDiagram(ggVenn_list[c(1,3)], label = "count")
```

# upset plots
```{r}
Upset_DET <- c(ggVenn_list$`Isoseq_DET_Ensembl&Pacbio`, ggVenn_list$`RNAseq_DET_Ensembl&Pacbio`, ggVenn_list$`RNAseq_DET_Ensembl&Pacbio_Kallistio`) %>% 
  unique() %>% 
  as.data.frame() %>%
  set_colnames("ID")

Upset_DET$`Isoseq_DET`[Upset_DET$ID %in% c(ggVenn_list$`Isoseq_DET_Ensembl&Pacbio`)] <- 1
Upset_DET$`Isoseq_DET`[is.na(Upset_DET$`Isoseq_DET`)] <- 0

Upset_DET$`RNAseq_DET`[Upset_DET$ID %in% c(ggVenn_list$`RNAseq_DET_Ensembl&Pacbio`)] <- 1
Upset_DET$`RNAseq_DET`[is.na(Upset_DET$`RNAseq_DET`)] <- 0

Upset_DET$`RNAseq_DET_Kallistio`[Upset_DET$ID %in% c(ggVenn_list$`RNAseq_DET_Ensembl&Pacbio_Kallistio`)] <- 1
Upset_DET$`RNAseq_DET_Kallistio`[is.na(Upset_DET$`RNAseq_DET_Kallistio`)] <- 0



upset(Upset_DET[,c("Isoseq_DET","RNAseq_DET_Kallistio")], nintersects = 30, mb.ratio = c(0.5, 0.5),main.bar.color = "blue")

```


