---
title: "DEG RNAseq HTSEQ"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(magrittr)
library(dplyr)
library(data.table)
library(limma)
library(edgeR)
library(ape)
library(readr)
library(tibble)
'%!in%' <- function(x,y){!('%in%'(x,y))}
```

# R featureCounts
```{r}
load("Inputs/counts-liver-brahman-withY.rda")
load("Inputs/geneannotation-brahman-withY.rda")
```

```{r}
# filter for 6 samples
gene_counts <- list()
gene_counts$counts <- counts$counts[,c("X7_sorted_merged.bam","X22_sorted_merged.bam","X53_sorted_merged.bam","X60_sorted_merged.bam","X65_sorted_merged.bam","X99_sorted_merged.bam")]

gene_counts$targets <- c("X7_sorted_merged.bam","X22_sorted_merged.bam","X53_sorted_merged.bam","X60_sorted_merged.bam","X65_sorted_merged.bam","X99_sorted_merged.bam")

gene_counts$stat <- counts$stat[,c("X7_sorted_merged.bam","X22_sorted_merged.bam","X53_sorted_merged.bam","X60_sorted_merged.bam","X65_sorted_merged.bam","X99_sorted_merged.bam")]
```

## Make master DGEList 

```{r}
x <- list()
x$counts = gene_counts$counts

colnames(x$counts) <-  gsub("\\_sorted_merged.bam", "", colnames(x$counts))
colnames(x$counts) <-  gsub("X", "", colnames(x$counts))
colnames(x$counts) <-  paste0("F", "", colnames(x$counts))

newanno <- newanno[c(1,4,5,7,9,10,11,12)] %>%
  set_colnames(c("seqnames","start","end","strand","geneID","entrze_ID","gene_name","type"))

x$genes <- counts$annotation
x$anno <- newanno[match(x$genes[,1],newanno[,5]),]
rownames(x$anno) <- c()
```


```{r}
#set expression level cutoff, can change from 0.5 to 1. 
sel <- rowSums(cpm(x$counts) > 0.5) >= 3
x$counts <- x$counts[sel,]
x$genes <- column_to_rownames(x$genes,"GeneID")[rownames(x$counts),] %>%
  tibble::rownames_to_column("GeneID")

```

```{r}
sampleinfo<- read.csv("Inputs/sampleinfo.csv")
sampleinfo$Animal_ID <- paste0("F",gsub("_.*","",sampleinfo$Animal_ID))
```

```{r}
x$counts <- x$counts[,match(sampleinfo[,1],colnames(x$counts))]
colnames(x$counts)

Subject <- as.factor(sampleinfo[,2])
Tissues <- as.factor(sampleinfo[,3])
Cross <- as.factor(sampleinfo[,4])
sampleinfo$phenotype <- gsub("Bi","Brahman",Cross)
sampleinfo$phenotype <- gsub("Bt","Angus",sampleinfo$phenotype)

Phenotype <- as.factor(sampleinfo[,5])
levels(Phenotype) <- c("Angus", "Brahman")

x <- new("DGEList", x)
dim(x)
```

### Normalization 

```{r}
x <- calcNormFactors(x, method="TMM")
```

### Bi vs Bt
```{r}
#set design matrix
desCross<- model.matrix(~0+Phenotype)
xCross <- estimateDisp(x,desCross)

#plot biological coefficient variation
plotBCV(xCross)

sqrt(xCross$common.dispersion)

vCross <- voomWithQualityWeights(xCross,design=desCross,normalize.method = "none",
                            plot=T,col=as.numeric(Subject))

#give the control and treatment colour label
cols <- rep("red",6)
cols[Phenotype=="Angus"] <- "black"

#mds plot with batch included
plotMDS(vCross,label=Phenotype,col=cols,dim.plot=c(1,2),main="RNA-seq MDSplot")

colnames(desCross)=c("Brahman","Angus")

contrCross <- makeContrasts("Brahman-Angus"= Brahman-Angus,levels=desCross)

vfitCross <- lmFit(vCross)

colnames(vfitCross$coefficients) <- rownames(contrCross)

vfitCross <- contrasts.fit(vfitCross,contrCross)
vfitCross <- eBayes(vfitCross)

resultsCross <- decideTests(vfitCross,p.value=0.05)

summary(decideTests(vfitCross,p.value=0.05, lfc = 1))

DEG_Brahman_Angus <- topTable(vfitCross,n=Inf,coef=1,p.value=0.05,lfc = 1, adjust.method="fdr")

DEG_Brahman_Angus$regulation <- "up"
DEG_Brahman_Angus$regulation[DEG_Brahman_Angus$logFC < 0] <- "down"
```

```{r}
write_tsv(DEG_Brahman_Angus,"Alltissues_significant_DEG_0.05.tsv")
```

