---
title: "RNAseq feature count transcript"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(magrittr)
library(dplyr)
library(stringr)
library(data.table)
library(S4Vectors)
library(GenomicRanges)
library(textshape)
library(tibble)
'%!in%' <- function(x,y){!('%in%'(x,y))}
```

# R featureCounts
## transcript
```{r}
library(Rsubread)

sortbam <- dir(path = "./feature_count" ,pattern=".sorted.bam", full.names = T)
#sortbam <- dir(path = "." ,pattern=".sorted.bam", full.names = T)

countsensembl <- featureCounts(sortbam, annot.ext= "Bos_indicus_hybrid.UOA_Brahman_1.104.gtf", isGTFAnnotationFile=TRUE, GTF.featureType="transcript", GTF.attrType="transcript_id", isPairedEnd=TRUE, reportReads=NULL)

#saveRDS(countsensembl,"countsensembl.rds")
```

```{r}
library(Rsubread)
Dir <- "/Users/kellydren/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Raw_data/map_to_brahman/post-SQANTI-filter/"
path_GTF <- file.path(Dir,"hq.5merge.collapsed_classification.filtered_lite.gtf")
                            
sortbam <- dir(path = "./feature_count" ,pattern=".sorted.bam", full.names = T)
#sortbam <- dir(path = "." ,pattern=".sorted.bam", full.names = T)

countpacbio <- featureCounts(sortbam, annot.ext=path_GTF, isGTFAnnotationFile=TRUE, GTF.featureType="transcript", GTF.attrType="transcript_id", isPairedEnd=TRUE, reportReads=NULL)

#saveRDS(countpacbio,"countpacbio.rds")
```


```{r}
countpacbio <- readRDS("countpacbio.rds")
countpacbio$counts

countsensembl <- readRDS("countsensembl.rds")
countsensembl$counts

RfeatureCounts_transcript_results_prefilter <- rbind(countpacbio$counts, countsensembl$counts[,colnames(countpacbio$counts)])
saveRDS(RfeatureCounts_transcript_results_prefilter,"RfeatureCounts_transcript_results_prefilter.rds")

```

```{r}
# library(Rsubread)
# 
# sortbam <- dir(path = "./feature_count" ,pattern=".sorted.bam", full.names = T)
# #sortbam <- dir(path = "." ,pattern=".sorted.bam", full.names = T)
# 
# #isoseq_transcriptome_Brahman.combined <- rtracklayer::import('isoseq_transcriptome_Brahman.combined.gtf')
# 
# countsensembl <- featureCounts(sortbam[[1]], annot.ext="isoseq_transcriptome_Brahman.combined.gtf", isGTFAnnotationFile=TRUE, GTF.featureType="transcript", GTF.attrType="transcript_id", isPairedEnd=TRUE, reportReads=NULL)
# 
# saveRDS(countsensembl,"RfeatureCounts_transcript_results_prefilter.rds")
```


# filter
```{r}
sel <- rowSums(cpm(RfeatureCounts_transcript_results_prefilter) > 0.5) >= 3
RfeatureCounts_transcript_results_afterfilter <- RfeatureCounts_transcript_results_prefilter[sel,]

saveRDS(RfeatureCounts_transcript_results_afterfilter,"RfeatureCounts_transcript_results_afterfilter.rds")
```
