---
title: "Transcriptome characterization"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(readr)
library(magrittr)
library(dplyr)
library(data.table)
library(readr)
library(ggVennDiagram)
library(biomaRt)
library(textshape)
library(ThreeDRNAseq)
```

### CPAT

```{r}
associated_isoform_TPM_with_novel_afterfilter <- read_rds("associated_isoform_counts_with_novel_afterfilter.rds")
```

```{r}
kallisto_transcript_TPM <- readRDS("kallisto_transcript_TPM.rds") %>% as.data.frame()
kallisto_transcript_TPM %<>% column_to_rownames("ID")

# filter the TPM
kallisto_transcript_TPM <- kallisto_transcript_TPM[TPM.filter(kallisto_transcript_TPM,sample.n = 3, tpm.cut = 1),]
```



```{r}
Bos_indicus_hybrid.UOA_Brahman_1.104 <- rtracklayer::import('Bos_indicus_hybrid.UOA_Brahman_1.104.gtf') %>% 
  as.data.frame()
```

```{r}
OP_Brahman_Isoseq <- read_tsv("/Users/kellydren/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Lloyd_RNAseq_cattle_R/RNAseq_transcript_chara/OP_Brahman_Isoseq/OP_Brahman_Isoseq.ORF_prob.best.tsv")

table(OP_Brahman_Isoseq$Coding_prob < 0.9)
# FALSE   TRUE 
# 372024   3609 

OP_Brahman_Isoseq <- subset(OP_Brahman_Isoseq, Coding_prob > 0.9)

#sum(OP_Brahman_Isoseq$ORF)/374633
```

```{r}
OP_Brahman_Isoseq$Coding_prob %>% plot
```

```{r}
OP_Brahman_RNAseq <- read_tsv("/Users/kellydren/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Lloyd_RNAseq_cattle_R/RNAseq_transcript_chara/OP_Brahman_RNAseq/OP_Brahman_RNAseq.ORF_prob.best.tsv")
```
```{r}
OP_Brahman <- rbind(OP_Brahman_RNAseq,OP_Brahman_Isoseq)
```

```{r}
OP_Brahman <- subset(OP_Brahman, seq_ID %in% rownames(associated_isoform_TPM_with_novel_afterfilter))

sum(OP_Brahman$ORF)/dim(OP_Brahman)[1]
```

```{r}

OP_Brahman <- subset(OP_Brahman, seq_ID %in% rownames(kallisto_transcript_TPM))

sum(OP_Brahman$ORF)/dim(OP_Brahman)[1]
```

### CPC2

```{r}
CPC2_Isoseq_outputs <- read.table("RNAseq_transcript_chara/CPC2_ORF/CPC2_Isoseq_outputs.txt") %>%
  set_colnames(c("Isoseq_ID","transcript_length","peptide_length","Fickett_score","pI","ORF_integrity","coding_probability","label"))


isoform_novel_afterfilter <- associated_isoform_TPM_with_novel_afterfilter[rownames(associated_isoform_TPM_with_novel_afterfilter) %>% grep("PB",.),]
table(rownames(isoform_novel_afterfilter) %in% CPC2_Isoseq_outputs$Isoseq_ID)

CPC2_Isoseq_outputs <- subset(CPC2_Isoseq_outputs, Isoseq_ID %in% rownames(isoform_novel_afterfilter))

CPC2_Isoseq_outputs$label %>% table()

CPC2_Isoseq_outputs_coding <- subset(CPC2_Isoseq_outputs, label %in% "coding")
```

