---
title: "differential expressed Genes using Isoseq data"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(ggplot2)
library(magrittr)
library(dplyr)
library(stringr)
library(data.table)
library(textshape)
library(biomaRt)
library(limma)
library(edgeR)
library(easyGgplot2)
library(readr)
library(tibble)
'%!in%' <- function(x,y){!('%in%'(x,y))}
```

# reading the filtered counts

```{r}
associated_gene_counts_with_novel_afterfilter <- readRDS("associated_gene_counts_with_novel_afterfilter.rds")
associated_gene_counts_afterfilter <- readRDS("associated_gene_counts_afterfilter.rds")
```

## Make master DGEList 
### associated_gene_counts_with_novel_afterfilter
```{r}
x <- list()

# Set the count
x$counts <- associated_gene_counts_with_novel_afterfilter

x$anno <- rownames(associated_gene_counts_with_novel_afterfilter)

x$genes <- data.frame(gene_id = x$anno)
```

```{r}
# see distribution
counts_distribution_for_plot <- x$counts %>%
  rownames_to_column("associated_gene") %>%
  reshape2::melt() %>%
  set_colnames(c("gene","samples","count"))

counts_distribution_for_plot$count_log <- log(counts_distribution_for_plot$count)

ggplot2.histogram(data=counts_distribution_for_plot, xName='count_log', groupName='samples') +
  ylab("Number of genes") + 
  xlab("log(Number of counts)") + 
  theme_bw()
```

```{r}
sampleinfo<- read.csv("sampleinfo.csv")
sampleinfo$Fetus_ID <- paste0("FL","",sampleinfo$Fetus_ID)
sampleinfo$Fetus_ID <- gsub("77","7", sampleinfo$Fetus_ID)
```

```{r}
x$counts <- x$counts[,match(sampleinfo[,2],colnames(x$counts))]
colnames(x$counts)

Subject <- as.factor(sampleinfo[,2])
Tissues <- as.factor(sampleinfo[,3])
Cross <- as.factor(sampleinfo[,4])
sampleinfo$phenotype <- gsub("Bi","Brahman",Cross)
sampleinfo$phenotype <- gsub("Bt","Angus",sampleinfo$phenotype)

Phenotype <- as.factor(sampleinfo[,5])
levels(Phenotype) <- c("Angus", "Brahman")

x <- new("DGEList", x)
dim(x)
```

### Normalization 

```{r}
x <- calcNormFactors(x, method="TMM")
```

### Bi vs Bt
```{r}
#set design matrix
desCross<- model.matrix(~0+Phenotype)
xCross <- estimateDisp(x,desCross)

#plot biological coefficient variation
plotBCV(xCross)

sqrt(xCross$common.dispersion)

vCross <- voomWithQualityWeights(xCross,design=desCross,normalize.method = "none",
                            plot=T,col=as.numeric(Subject))

#give the control and treatment colour label
cols <- rep("red",6)
cols[Phenotype=="Angus"] <- "black"

#mds plot with batch included
plotMDS(vCross,label=Phenotype,col=cols,dim.plot=c(1,2),main="RNA-seq MDSplot")

colnames(desCross)=c("Brahman", "Angus")

contrCross <- makeContrasts("Brahman-Angus"= Brahman-Angus,levels=desCross)

vfitCross <- lmFit(vCross)

colnames(vfitCross$coefficients) <- rownames(contrCross)

vfitCross <- contrasts.fit(vfitCross,contrCross)
vfitCross <- eBayes(vfitCross)

resultsCross <- decideTests(vfitCross,p.value=0.05)

summary(decideTests(vfitCross,p.value=0.05, lfc = 1))

DEG_Brahman_Angus <- topTable(vfitCross,n=Inf,coef=1,p.value=0.05,lfc = 1,adjust.method="fdr")

DEG_Brahman_Angus$regulation <- "up"
DEG_Brahman_Angus$regulation[DEG_Brahman_Angus$logFC < 0] <- "down"
```

```{r}
write_tsv(DEG_Brahman_Angus,"IsoSeq_significant_with_novel_DEG_0.05.tsv")
```

### associated_gene_counts_afterfilter
```{r}
x <- list()

# Set the count
x$counts <- associated_gene_counts_afterfilter

x$anno <- rownames(associated_gene_counts_afterfilter)

x$genes <- data.frame(gene_id = x$anno)
```

```{r}
# see distribution
counts_distribution_for_plot <- x$counts %>%
  rownames_to_column("associated_gene") %>%
  reshape2::melt() %>%
  set_colnames(c("gene","samples","count"))

counts_distribution_for_plot$count_log <- log(counts_distribution_for_plot$count)

ggplot2.histogram(data=counts_distribution_for_plot, xName='count_log', groupName='samples') +
  ylab("Number of genes") + 
  xlab("log(Number of counts)") + 
  theme_bw()
```

```{r}
sampleinfo<- read.csv("sampleinfo.csv")
sampleinfo$Fetus_ID <- paste0("FL","",sampleinfo$Fetus_ID)
sampleinfo$Fetus_ID <- gsub("77","7", sampleinfo$Fetus_ID)
```

```{r}
x$counts <- x$counts[,match(sampleinfo[,2],colnames(x$counts))]
colnames(x$counts)

Subject <- as.factor(sampleinfo[,2])
Tissues <- as.factor(sampleinfo[,3])
Cross <- as.factor(sampleinfo[,4])
sampleinfo$phenotype <- gsub("Bi","Brahman",Cross)
sampleinfo$phenotype <- gsub("Bt","Angus",sampleinfo$phenotype)

Phenotype <- as.factor(sampleinfo[,5])
levels(Phenotype) <- c("Angus", "Brahman")

x <- new("DGEList", x)
dim(x)
```

### Normalization 

```{r}
x <- calcNormFactors(x, method="TMM")
```

### Bi vs Bt
```{r}
#set design matrix
desCross<- model.matrix(~0+Phenotype)
xCross <- estimateDisp(x,desCross)

#plot biological coefficient variation
plotBCV(xCross)

sqrt(xCross$common.dispersion)

vCross <- voomWithQualityWeights(xCross,design=desCross,normalize.method = "none",
                            plot=T,col=as.numeric(Subject))

#give the control and treatment colour label
cols <- rep("red",6)
cols[Phenotype=="Angus"] <- "black"

#mds plot with batch included
plotMDS(vCross,label=Phenotype,col=cols,dim.plot=c(1,2),main="RNA-seq MDSplot")

colnames(desCross)=c("Brahman", "Angus")

contrCross <- makeContrasts("Brahman-Angus"= Brahman-Angus,levels=desCross)

vfitCross <- lmFit(vCross)

colnames(vfitCross$coefficients) <- rownames(contrCross)

vfitCross <- contrasts.fit(vfitCross,contrCross)
vfitCross <- eBayes(vfitCross)

resultsCross <- decideTests(vfitCross,p.value=0.05)

summary(decideTests(vfitCross,p.value=0.05, lfc = 1))

DEG_Brahman_Angus <- topTable(vfitCross,n=Inf,coef=1,p.value=0.05,lfc = 1,adjust.method="fdr")

DEG_Brahman_Angus$regulation <- "up"
DEG_Brahman_Angus$regulation[DEG_Brahman_Angus$logFC < 0] <- "down"
```

```{r}
write_tsv(DEG_Brahman_Angus,"IsoSeq_significant_without_novel_DEG_0.05.tsv")
```
