---
title: "differential transcripts"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(ggplot2)
library(magrittr)
library(dplyr)
library(stringr)
library(data.table)
library(textshape)
library(biomaRt)
library(limma)
library(edgeR)
library(rtracklayer)
'%!in%' <- function(x,y){!('%in%'(x,y))}
```

```{r}
#Brahman pre-SQANTI 
Dir <- "/Users/kellydren/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Raw_data/map_to_brahman/post-SQANTI-filter/"
classification <- read.delim(file.path(Dir,"hq.5merge.collapsed_classification.filtered_lite_classification.txt"),
                             stringsAsFactors = FALSE)

```

```{r}
#Remove unwanted variables
classification.keep <- classification %>% 
  dplyr::select(isoform,FL.F22,FL.F99,FL.F65,FL.F53,FL.F60,FL.F77)

#remove single exons transcripts
#classification.keep <- classification.keep %>% dplyr::filter(exons != 1)
head(classification.keep)
```

# make tables for tappAS

```{r}
classification <- data.frame(SeqName = classification$isoform,
           Start = classification$CDS_start,
           End = classification$CDS_end,
           Score =  ".",
           Strand = classification$strand,
           Phase =  ".",
           Source = "Pacbio",
           Feature = "coding",
           Attributes = classification$associated_gene)

classification <- classification[!is.na(classification$Start),]

classification_Grange <- makeGRangesFromDataFrame(classification)
mcols(classification_Grange) <- classification
rtracklayer::export(classification_Grange,"Iso_seq_annotation_file_for_tappAS.gff3",format = "gff3")

classification.keep <- classification.keep %>%
  set_colnames(c("isoform","F22","F53","F60","F65","F7","F99")) %>% 
  column_to_rownames("isoform")

classification.keep <- classification.keep[classification_Grange$SeqName,]

write_tsv(classification.keep %>% 
  rownames_to_column("isoform"), "Iso_seq_espression_matrix_for_tappAS.tsv")

sampleinfo<- read.csv("sampleinfo.csv")
sampleinfo

sampleinfo_tappAS <- sampleinfo[,c(2,4)] %>%
  set_colnames(c("Sample","Group"))

sampleinfo_tappAS$Group <- gsub("Bi","Brahman",sampleinfo_tappAS$Group) 
sampleinfo_tappAS$Group  <- gsub("Bt","Angus",sampleinfo_tappAS$Group)
sampleinfo_tappAS$Sample  <- gsub("77","7",sampleinfo_tappAS$Sample)
sampleinfo_tappAS$Sample <- paste0("F",sampleinfo_tappAS$Sample)

sampleinfo_tappAS %>%
  write_tsv("Iso_seq_design_for_tappAS.tsv")

```
