---
title: "Transcript structures (for PB.35887)"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(magrittr)
library(dplyr)
library(stringr)
library(readr)
library(tibble)
library(DRIMSeq)
library(tidyr)
library(ggplot2)
library(ggtranscript)
library(matrixStats)
library(ThreeDRNAseq)
library(reshape2)

'%!in%' <- function(x,y){!('%in%'(x,y))}
```

# RNA-seq
```{r}
kallisto_transcript_TPM <- readRDS("~/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Lloyd_RNAseq_cattle_R/kallisto_transcript_TPM.rds")
```

```{r}
kallisto_transcript_TPM %<>% column_to_rownames("ID")

kallisto_transcript_TPM <- kallisto_transcript_TPM[TPM.filter(kallisto_transcript_TPM,sample.n = 3, tpm.cut = 1),]
```

# Iso-seq
```{r}
#Brahman post-SQANTI 
Dir <- "/Users/kellydren/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Raw_data/map_to_brahman/post-SQANTI-filter/"

# inport GTF to match gene_id
hq.5merge.collapsed_classification.filtered_lite <- rtracklayer::import(file.path(Dir,'hq.5merge.collapsed_classification.filtered_lite.gtf')) %>%
  as.data.frame()

associated_isoform_counts_with_novel_afterfilter <- readRDS("associated_isoform_counts_with_novel_afterfilter.rds")
```

# PB.35887
```{r}
Transcripts_in_ENSBIXG00005004010 <- hq.5merge.collapsed_classification.filtered_lite %>% subset(gene_id %in% "PB.35887")
Transcripts_in_ENSBIXG00005004010 <- subset(Transcripts_in_ENSBIXG00005004010, type %in% "exon")

```

```{r}
gene_to_plot <- associated_isoform_counts_with_novel_afterfilter[as.character(unique(Transcripts_in_ENSBIXG00005004010$transcript_id)),] %>% drop_na()

gene_to_plot <- gene_to_plot %>%
  rownames_to_column("isoform") %>%
  melt()

ggplot(data=gene_to_plot, aes(x=variable, y=value, fill=isoform)) +
  geom_bar(stat="identity", width=0.8) +  
 geom_text(aes(label=isoform), vjust=1.6, color="white",
          position = position_dodge(0.9), size=3.5) +
  theme_bw()

ggplot(data=gene_to_plot_v1, aes(x=isoform, y=value, fill=variable)) +
    geom_bar(stat="identity", position=position_dodge()) +  
    geom_text(aes(label=isoform), vjust=1.6, color="white",
              position = position_dodge(0.9), size=3.5)
```

```{r}
gene_to_plot <- associated_isoform_counts_with_novel_afterfilter[as.character(unique(Transcripts_in_ENSBIXG00005004010$transcript_id)),] %>% drop_na()


Rank  <- abs(rowMeans2(as.matrix(gene_to_plot[,c(1:3)])) - rowMeans2(as.matrix(gene_to_plot[,c(4:6)]))) %>%
  as.data.frame() %>%
  set_colnames("Diff")

Rank$regulation <- (rowMeans2(as.matrix(gene_to_plot[,c(1:3)])) - rowMeans2(as.matrix(gene_to_plot[,c(4:6)])))>0
Rank$ID <- rownames(gene_to_plot)


Rank$Rank <- c(1:nrow(Rank))

Rank <- Rank %>%
  arrange(Diff)

heatmap(as.matrix(gene_to_plot[Rank$Rank,]), Rowv = NA)

heatmap(as.matrix(gene_to_plot[Rank$Rank,]), Rowv = NA, scale="column")
```

```{r}

gene_to_plot <- rownames_to_column(gene_to_plot, "ID")
gene_to_plot_SR <- kallisto_transcript_TPM[as.character(unique(Transcripts_in_ENSBIXG00005004010$transcript_id)),] %>% drop_na()

gene_to_plot_SR <- rownames_to_column(gene_to_plot_SR, "ID")
gene_to_plot <- as.data.frame(intersect(gene_to_plot_SR$ID,gene_to_plot$ID)) %>%
  set_colnames("ID") %>%
  left_join(gene_to_plot)%>%
  left_join(gene_to_plot_SR)

# gene_to_plot <- as.data.frame(gene_to_plot_SR$ID,gene_to_plot$ID) %>%
#   set_colnames("ID") %>%
#   left_join(gene_to_plot)%>%
#   left_join(gene_to_plot_SR)

gene_to_plot %<>% column_to_rownames("ID")
#
# Rank  <- abs(rowMeans2(as.matrix(gene_to_plot[,c(7,12,10)])) - rowMeans2(as.matrix(gene_to_plot[,c(8,9,11)]))) %>%
#   as.data.frame() %>%
#   set_colnames("Diff")
#
# Rank$regulation <- (rowMeans2(as.matrix(gene_to_plot[,c(7,12,10)])) - rowMeans2(as.matrix(gene_to_plot[,c(8,9,11)])))>0
# Rank$ID <- rownames(gene_to_plot)
#
#
# Rank$Rank <- c(1:nrow(Rank))
#
# Rank <- Rank %>%
#   arrange(Diff)

gene_to_plot %<>%
  set_colnames(c("isoseq_22","isoseq_99", "isoseq_65","isoseq_53","isoseq_60","isoseq_7","illumina_22","illumina_53","illumina_60","illumina_65","illumina_7","illumina_99"))

gene_to_plot <- gene_to_plot[,c("isoseq_22","isoseq_99", "isoseq_65","isoseq_53","isoseq_60","isoseq_7","illumina_22","illumina_99","illumina_65","illumina_53","illumina_60","illumina_7")]

heatmap(as.matrix(gene_to_plot[Rank$ID,]), Colv = NA, Rowv = NA)

heatmap(as.matrix(gene_to_plot[Rank$ID,]),Colv = NA, Rowv = NA, scale="column")
legend(x="right", legend=c("min", "med", "max"),fill=heat.colors(3))

```



```{r}
Transcripts_in_ENSBIXG00005004010 <- subset(Transcripts_in_ENSBIXG00005004010, transcript_id %in% c(rownames(gene_to_plot[Rank$Rank,])))
Transcripts_in_ENSBIXG00005004010$transcript_id <- factor(Transcripts_in_ENSBIXG00005004010$transcript_id, levels = c(rownames(gene_to_plot[Rank$Rank,])))

Transcripts_in_ENSBIXG00005004010$Dominant <- "Angus"
Transcripts_in_ENSBIXG00005004010$Dominant[Transcripts_in_ENSBIXG00005004010$transcript_id %in% Rank$ID[Rank$regulation]] <- "Brahman"


Transcripts_in_ENSBIXG00005004010 %>%
    ggplot(aes(
        xstart = start,
        xend = end,
        y = transcript_id)) +
    geom_range(
        aes(fill = Dominant)) +
    geom_intron(
        data = to_intron(Transcripts_in_ENSBIXG00005004010, "transcript_id"),
        aes(strand = strand)) +
  scale_fill_manual(values = c("#0076d3", "#ff00ab")) +
  theme_bw() 
```



# sessionInfo
```{r}
sessionInfo()
```

