---
title: "Check if the overlap gene exist in groups of DEGs"
author: "Kelly Ren"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2: default
bibliography: FinalthesisRef.bib
link-citations: yes
fig_caption: yes
toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(magrittr)
library(readr)
library(dplyr)
library(stringr)
library(ggplot2)
library(GenomicRanges)

'%!in%' <- function(x,y)!('%in%'(x,y))
```

```{r}
setwd("~/Documents/Kelly_2021/Research/Lloyd_PacBio_ILMN_quantification/Lloyd_RNAseq_cattle_R")
```

```{r}
Bos_indicus_hybrid.UOA_Brahman_1.104 <- rtracklayer::import('Bos_indicus_hybrid.UOA_Brahman_1.104.gtf') %>% subset(type %in% "gene")

# overlap all the genes
overlapped_genes <- findOverlaps(Bos_indicus_hybrid.UOA_Brahman_1.104,Bos_indicus_hybrid.UOA_Brahman_1.104)

# remove overlap with itself
overlapped_genes <- overlapped_genes[!(queryHits(overlapped_genes) == subjectHits(overlapped_genes)),]

# keep one between AB BA overlaps
cob_query_subject <- paste0(queryHits(overlapped_genes),"_", subjectHits(overlapped_genes))
re_cob_query_subject <- paste0(subjectHits(overlapped_genes),"_", queryHits(overlapped_genes))

for(x in c(1:length(cob_query_subject))){

  if(cob_query_subject[x] %in% re_cob_query_subject){

    digit <- strsplit(cob_query_subject[x],"_") %>%
      unlist()

    re_cob_query_subject[grep(paste0("\\b",digit[2],"_",digit[1],"\\b"), re_cob_query_subject)] <- NA}
}

overlapped_genes <- overlapped_genes[is.na(re_cob_query_subject)]
```

# in any
```{r}
# calculate all_overlapped_genes
query_Hits <- Bos_indicus_hybrid.UOA_Brahman_1.104[queryHits(overlapped_genes),] %>%
  as.data.frame()
dim(query_Hits)

subject_Hits <- Bos_indicus_hybrid.UOA_Brahman_1.104[subjectHits(overlapped_genes),] %>%
  as.data.frame()
dim(subject_Hits)

all_overlapped_genes <- rbind(query_Hits, subject_Hits) %>%
  unique()
```

```{r}
both <- read_tsv("DEG_both_RNAseq_Isoseq_coordinates.tsv")
Isoseq_only <- read_tsv("DEG_Isoseq_only_coordinates.tsv", col_names = F)
RNAseq_only <- read_tsv("DEG_RNAseq_only_coordinates.tsv")

summary_overlap <- data_frame(both = NA,
           Isoseq_only = NA,
           RNAseq_only = NA)

# % of the overlap genes in both
summary_overlap$both <- table(both$gene_id %in% all_overlapped_genes$gene_id)[2]/length(unique(both$gene_id))

# % of the overlap genes in Isoseq_only
summary_overlap$Isoseq_only <- table(Isoseq_only$X5 %in% all_overlapped_genes$gene_id)[2]/length(unique(Isoseq_only$X5))

# % of the overlap genes in RNAseq_only
summary_overlap$RNAseq_only <- table(RNAseq_only$gene_id %in% all_overlapped_genes$gene_id)[2]/length(unique(RNAseq_only$gene_id))

summary_overlap_all <- summary_overlap * 100

reshape2::melt(summary_overlap_all) %>%
  ggplot(aes(x=variable, y=value)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_bw()
```

# in 25%
```{r}
all_overlapped_genes_grange <- makeGRangesFromDataFrame(all_overlapped_genes)
mcols(all_overlapped_genes_grange) <- all_overlapped_genes[,-c(1:5)]
# any overlapping
overlapped_genes_any <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="any") %>%
  as.data.frame()

overlapped_genes_any$type <- paste0(overlapped_genes_any$queryHits,"_",overlapped_genes_any$subjectHits)

# equal overlapping
overlapped_genes_equal <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal") %>%
  as.data.frame()

overlapped_genes_equal$type <- paste0(overlapped_genes_equal$queryHits,"_",overlapped_genes_equal$subjectHits)

# subset the situations of overlapping that we need
overlapped_genes <- subset(overlapped_genes_any, !(type %in% overlapped_genes_equal$type))
dim(overlapped_genes)
# 4816    3
```

```{r}
# find in percentage (25%)
overlaps <- pintersect(all_overlapped_genes_grange[overlapped_genes$queryHits], Bos_indicus_hybrid.UOA_Brahman_1.104[overlapped_genes$subjectHits])

percentOverlap <- width(overlaps) / width(all_overlapped_genes_grange[overlapped_genes$queryHits])
overlapped_genes <- overlapped_genes[percentOverlap > 0.25,]

# overlap genes
overlapped_genes <- all_overlapped_genes_grange[overlapped_genes$queryHits,]
overlapped_genes <- unique(overlapped_genes)

length(overlapped_genes)
# 2392
```


```{r}
# overlap all the genes
overlapped_genes <- findOverlaps(overlapped_genes,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal")

# remove overlap with itself
overlapped_genes <- Bos_indicus_hybrid.UOA_Brahman_1.104[subjectHits(overlapped_genes),]
```

```{r}
summary_overlap <- data_frame(both = 0,
           Isoseq_only = 0,
           RNAseq_only = 0)

# % of the overlap genes in both
summary_overlap$both <- table(both$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(both$gene_id))

# % of the overlap genes in Isoseq_only
summary_overlap$Isoseq_only <- table(Isoseq_only$X5 %in% overlapped_genes$gene_id)[2]/length(unique(Isoseq_only$X5))

# % of the overlap genes in RNAseq_only
summary_overlap$RNAseq_only <- table(RNAseq_only$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(RNAseq_only$gene_id))

summary_overlap[is.na(summary_overlap)] <- 0
summary_overlap_25 <- summary_overlap

summary_overlap_25$Isoseq_only <- summary_overlap_25$Isoseq_only * 100
summary_overlap_25$RNAseq_only <- summary_overlap_25$RNAseq_only * 100

```

# in 50%

```{r}
all_overlapped_genes_grange <- makeGRangesFromDataFrame(all_overlapped_genes)
mcols(all_overlapped_genes_grange) <- all_overlapped_genes[,-c(1:5)]

# any overlapping
overlapped_genes_any <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="any") %>%
  as.data.frame()

overlapped_genes_any$type <- paste0(overlapped_genes_any$queryHits,"_",overlapped_genes_any$subjectHits)

# equal overlapping
overlapped_genes_equal <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal") %>%
  as.data.frame()

overlapped_genes_equal$type <- paste0(overlapped_genes_equal$queryHits,"_",overlapped_genes_equal$subjectHits)

# subset the situations of overlapping that we need
overlapped_genes <- subset(overlapped_genes_any, !(type %in% overlapped_genes_equal$type))

```

```{r}
# find in percentage (50%)
overlaps <- pintersect(all_overlapped_genes_grange[overlapped_genes$queryHits], Bos_indicus_hybrid.UOA_Brahman_1.104[overlapped_genes$subjectHits])

percentOverlap <- width(overlaps) / width(all_overlapped_genes_grange[overlapped_genes$queryHits])
overlapped_genes <- overlapped_genes[percentOverlap > 0.50,]

# overlap genes
overlapped_genes <- all_overlapped_genes_grange[overlapped_genes$queryHits,]
overlapped_genes <- unique(overlapped_genes)

length(overlapped_genes)
# 2256
```


```{r}
# overlap all the genes
overlapped_genes <- findOverlaps(overlapped_genes,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal")

# remove overlap with itself
overlapped_genes <- Bos_indicus_hybrid.UOA_Brahman_1.104[subjectHits(overlapped_genes),]
```

```{r}
summary_overlap <- data_frame(both = NA,
           Isoseq_only = NA,
           RNAseq_only = NA)

# % of the overlap genes in both
summary_overlap$both <- table(both$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(both$gene_id))

# % of the overlap genes in Isoseq_only
summary_overlap$Isoseq_only <- table(Isoseq_only$X5 %in% overlapped_genes$gene_id)[2]/length(unique(Isoseq_only$X5))

# % of the overlap genes in RNAseq_only
summary_overlap$RNAseq_only <- table(RNAseq_only$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(RNAseq_only$gene_id))

summary_overlap[is.na(summary_overlap)] <- 0
summary_overlap_50 <- summary_overlap

summary_overlap_50$Isoseq_only <- summary_overlap_50$Isoseq_only * 100
summary_overlap_50$RNAseq_only <- summary_overlap_50$RNAseq_only * 100

```

# in 75%
```{r}
all_overlapped_genes_grange <- makeGRangesFromDataFrame(all_overlapped_genes)
mcols(all_overlapped_genes_grange) <- all_overlapped_genes[,-c(1:5)]

# any overlapping
overlapped_genes_any <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="any") %>%
  as.data.frame()

overlapped_genes_any$type <- paste0(overlapped_genes_any$queryHits,"_",overlapped_genes_any$subjectHits)

# equal overlapping
overlapped_genes_equal <- findOverlaps(all_overlapped_genes_grange,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal") %>%
  as.data.frame()

overlapped_genes_equal$type <- paste0(overlapped_genes_equal$queryHits,"_",overlapped_genes_equal$subjectHits)

# subset the situations of overlapping that we need
overlapped_genes <- subset(overlapped_genes_any, !(type %in% overlapped_genes_equal$type))

```


```{r}
# find in percentage (75%)
overlaps <- pintersect(all_overlapped_genes_grange[overlapped_genes$queryHits], Bos_indicus_hybrid.UOA_Brahman_1.104[overlapped_genes$subjectHits])

percentOverlap <- width(overlaps) / width(all_overlapped_genes_grange[overlapped_genes$queryHits])
overlapped_genes <- overlapped_genes[percentOverlap > 0.75,]

# overlap genes
overlapped_genes <- all_overlapped_genes_grange[overlapped_genes$queryHits,]
overlapped_genes <- unique(overlapped_genes)

length(overlapped_genes)
# 2176
```


```{r}
# overlap all the genes
overlapped_genes <- findOverlaps(overlapped_genes,Bos_indicus_hybrid.UOA_Brahman_1.104, type="equal")

# remove overlap with itself
overlapped_genes <- Bos_indicus_hybrid.UOA_Brahman_1.104[subjectHits(overlapped_genes),]
```

```{r}
summary_overlap <- data_frame(both = NA,
           Isoseq_only = NA,
           RNAseq_only = NA)

# % of the overlap genes in both
summary_overlap$both <- table(both$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(both$gene_id))

# % of the overlap genes in Isoseq_only
summary_overlap$Isoseq_only <- table(Isoseq_only$X5 %in% overlapped_genes$gene_id)[2]/length(unique(Isoseq_only$X5))

# % of the overlap genes in RNAseq_only
summary_overlap$RNAseq_only <- table(RNAseq_only$gene_id %in% overlapped_genes$gene_id)[2]/length(unique(RNAseq_only$gene_id))

summary_overlap[is.na(summary_overlap)] <- 0
summary_overlap_75 <- summary_overlap

summary_overlap_75$Isoseq_only <- summary_overlap_75$Isoseq_only * 100
summary_overlap_75$RNAseq_only <- summary_overlap_75$RNAseq_only * 100

```


```{r}
summary_overlap <- summary_overlap_25 %>%
  rbind(summary_overlap_50) %>%
  rbind(summary_overlap_75) %>%
  set_rownames(c("25% cor","50% cor","75% cor")) %>%
  tibble::rownames_to_column("coverage")

reshape2::melt(summary_overlap) %>%
  ggplot(aes(x=variable, y=value, fill = coverage)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_bw()

```


```{r}
# Grange of overlap genes in each group

## both (4/33 = 0.1212121)
both_Grange <- subset(all_overlapped_genes, gene_id %in% both$gene_id)
both_gene_lengh <- both_Grange$width %>%
  as.data.frame() %>%
  set_colnames("gene_lengh")
  
both_gene_lengh$gene_id <- both_Grange$gene_id
both_gene_lengh %<>% unique()

rownames(both_gene_lengh) <- NULL
both_gene_lengh %<>% tibble::column_to_rownames("gene_id")

#                    gene_lengh
# ENSBIXG00005018766     135371
# ENSBIXG00005012404     207680
# ENSBIXG00005008802     385672
# ENSBIXG00005006062     134564

## Isoseq_only (23/124 = 0.1854839)
Isoseq_only_Grange <- subset(all_overlapped_genes, gene_id %in% Isoseq_only$X5)

Isoseq_only_gene_lengh <- Isoseq_only_Grange$width %>%
  as.data.frame() %>%
  set_colnames("gene_lengh")
  
Isoseq_only_gene_lengh$gene_id <- Isoseq_only_Grange$gene_id
Isoseq_only_gene_lengh %<>% unique()

rownames(Isoseq_only_gene_lengh) <- NULL
Isoseq_only_gene_lengh %<>% tibble::column_to_rownames("gene_id")

## RNAseq_only (18/203 = 0.08866995)
RNAseq_only_Grange <- subset(all_overlapped_genes, gene_id %in% RNAseq_only$gene_id)

RNAseq_only_gene_lengh <- RNAseq_only_Grange$width %>%
  as.data.frame() %>%
  set_colnames("gene_lengh")
  
RNAseq_only_gene_lengh$gene_id <- RNAseq_only_Grange$gene_id
RNAseq_only_gene_lengh %<>% unique()

rownames(RNAseq_only_gene_lengh) <- NULL
RNAseq_only_gene_lengh %<>% tibble::column_to_rownames("gene_id")
```

```{r}
# Grange of overlaps for those genes
# one gene might have multiple overlaps and the length of those overlaps are different

overlaps_both_Grange <- subset(overlaps, gene_id %in% both$gene_id)
overlaps_Isoseq_only_Grange <- subset(overlaps, gene_id %in% Isoseq_only$X5)
overlaps_RNAseq_only_Grange <- subset(overlaps, gene_id %in% RNAseq_only$gene_id)

# Calculate max how much of one gene has been overlapped 
both_value <- lapply(c(1:nrow(both_gene_lengh)), function(x, Gene_length = both_gene_lengh, overlaps_Grange =  overlaps_both_Grange){
  
  # pull out all the overlapps for one gene 
  max_cor <- subset(overlaps_Grange, gene_id %in% rownames(Gene_length)[[x]]) %>%
    width
  # pick the max one
  max_cor <-  max_cor %>% max()
  
  # calculate how much percentage the max coverage out of the gene length
  (max_cor/Gene_length$gene_lengh[[x]])*100
  
}) %>% 
  unlist() %>%
  as.data.frame() %>%
  set_rownames(rownames(both_gene_lengh)) %>%
  set_colnames("value")

both_value
```

```{r}
# applied the same thing to group RNAseq_only and Isoseq_only
Isoseq_only_value <- lapply(c(1:nrow(Isoseq_only_gene_lengh)), function(x, Gene_length = Isoseq_only_gene_lengh, overlaps_Grange =  overlaps_Isoseq_only_Grange){
  
  # pull out all the overlapps for one gene
  max_cor <- subset(overlaps_Grange, gene_id %in% rownames(Gene_length)[[x]]) %>%
    width
  
  # remove the self-overlapps and pick the max one
  max_cor <- max_cor%>% max()
  
  # calculate how much percentage the max coverage out of the gene length
  (max_cor/Gene_length$gene_lengh[[x]])*100
}) %>% 
  unlist() %>%
  as.data.frame() %>%
  set_rownames(rownames(Isoseq_only_gene_lengh)) %>%
  set_colnames("value")

Isoseq_only_value
```

```{r}
# applied the same thing to group RNAseq_only and Isoseq_only
RNAseq_only_value <- lapply(c(1:nrow(RNAseq_only_gene_lengh)), function(x, Gene_length = RNAseq_only_gene_lengh, overlaps_Grange = overlaps_RNAseq_only_Grange){
  
  # pull out all the overlapps for one gene
  max_cor <- subset(overlaps_Grange, gene_id %in% rownames(Gene_length)[[x]]) %>%
  width
  
  # remove the self-overlapps and pick the max one
  max_cor <- max_cor %>% max()

  # calculate how much percentage the max coverage out of the gene length
  (max_cor/Gene_length$gene_lengh[[x]])*100
}) %>% 
  unlist() %>%
  as.data.frame() %>%
  set_rownames(rownames(RNAseq_only_gene_lengh)) %>%
  set_colnames("value")

RNAseq_only_value
```

```{r}
# put the vaule back with 0 for other DEGs have no overlap 

both_value <- both$gene_id[!(both$gene_id %in% rownames(both_value))] %>%
  as.data.frame() %>%
  mutate(value = 0 ) %>%
  tibble::column_to_rownames(".") %>%
  rbind(both_value) %>% 
  tibble::rownames_to_column("gene_id") %>%
  mutate(Group = "both")
  
Isoseq_only_value <- Isoseq_only$X5[!(Isoseq_only$X5 %in% rownames(Isoseq_only_value))] %>%
  as.data.frame() %>%
  mutate(value = 0 ) %>%
  tibble::column_to_rownames(".") %>%
  rbind(Isoseq_only_value) %>% 
  tibble::rownames_to_column("gene_id") %>%
  mutate(Group = "Isoseq_only")

RNAseq_only_value <- RNAseq_only$gene_id[!(RNAseq_only$gene_id %in% rownames(RNAseq_only_value))] %>%
  as.data.frame() %>%
  mutate(value = 0 ) %>%
  tibble::column_to_rownames(".") %>%
  rbind(RNAseq_only_value) %>% 
  tibble::rownames_to_column("gene_id") %>%
  mutate(Group = "RNAseq_only")

summary_overlap_values <- rbind(both_value,Isoseq_only_value) %>%
  rbind(RNAseq_only_value)
```

```{r}
ggplot(summary_overlap_values ,aes(x=Group, y=value, color=Group, shape = Group, size = 1)) +
  geom_boxplot(notch=TRUE,outlier.shape = NA) +
  geom_jitter() +
  scale_color_manual(values=c("#00c2c6","#f9b410","#ff6c67")) +
  theme_bw()

```


```{r}
wilcox.test(subset(summary_overlap_values, Group %in% "both")$value,subset(summary_overlap_values, Group %in% "Isoseq_only")$value)

wilcox.test(subset(summary_overlap_values, Group %in% "both")$value,subset(summary_overlap_values, Group %in% "RNAseq_only")$value)

wilcox.test(subset(summary_overlap_values, Group %in% "Isoseq_only")$value,subset(summary_overlap_values, Group %in% "RNAseq_only")$value)
```

# sessionInfo
```{r}
sessionInfo()
```

